sdk/trace/provider.go

```diff
@@ -115,8 +115,9 @@ func (p *TracerProvider) Tracer(name string, opts ...trace.TracerOption) trace.T
 		name = defaultTracerName
 	}
 	il := instrumentation.Library{
-		Name:    name,
-		Version: c.InstrumentationVersion,
+		Name:      name,
+		Version:   c.InstrumentationVersion,
+		SchemaURL: c.SchemaURL,
```

Comment:
Should this handle an empty `c.SchemaURL` at this point and insert the schema supported by the SDK?

Reply:
No, I think it should not. The Schema URL needs to be explicitly provided. The choice of the Tracer schema belongs to the instrumentation library. If the instrumentation library did not provide a schema URL, it really means the library does not know, does not care or is it too old be aware of what is a schema URL, so we need to keep the field empty.

As opposed to the above, the schema URL that the SDK implements _should_ be applied to auto-detected Resources, because it is the SDK's choice in that case, the instrumentation code actually lives in the SDK. But for the Tracer the SDK cannot make that choice.

(I am now looking at our https://github.com/open-telemetry/opentelemetry-go-contrib to understand how we can make it easy for instrumentation libraries to provide the correct schema URL when they get a Tracer, such that the semantic conventions they later use when creating spans with that Tracer correspond correctly to the schema URL).